FilePersistence/src/version_control/merge/ChangeGraph.cpp

```diff
@@ -448,14 +446,105 @@ void ChangeGraph::removeLabelConflictsBetweenChildren(Model::NodeIdType parentId
 	}
 }
 
-void ChangeGraph::updateIncomingLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/)
+void ChangeGraph::updateListChanges(Model::NodeIdType parentId, IdToLabelMap labelMap, GenericTree* baseTree)
 {
-	Q_ASSERT(false);
-}
+	for (auto nodeId : labelMap.keys())
+	{
+		QString labelA;
+		QString labelB;
+		QString labelNone;
+		auto labels = labelMap.values(nodeId);
+		for (auto label : labels)
+		{
+			if (label.branch_.testFlag(MergeChange::BranchA))	labelA = label.label_;
+			else if (label.branch_.testFlag(MergeChange::BranchB))	labelB = label.label_;
+			else labelNone = label.label_;	//Base
+		}
 
-void ChangeGraph::updateOutgoingLabels(Model::NodeIdType /*parentId*/, IdToLabelMap /*labelMap*/)
-{
-	Q_ASSERT(false);
+		// Need to adjust Update Changes as well
+		auto parentNode = baseTree->find(parentId);
+		Q_ASSERT(parentNode != NULL);
+		auto childNode = baseTree->find(nodeId);
+		if (childNode && parentNode->children().contains(childNode))	// Deletion, Move Out, LabelChange, Stable
+		{
+			auto changeIt = changesForNode_.find(nodeId);
+			while (changeIt != changesForNode_.end() && changeIt.key() == nodeId)
+			{
+				// Ignore stationary changes
+				if (changeIt.value()->type() == ChangeType::Stationary)
+				{
+					// Since label changes are already deleted
+					Q_ASSERT(changeIt.value()->updateFlags().testFlag(ChangeDescription::Value) ||
+								changeIt.value()->updateFlags().testFlag(ChangeDescription::Type));
+					continue;
+				}
+
+				// Must be DELETION / MOVE OUT change because it is present in base
+				// and stationary changes are ignored
+				Q_ASSERT(changeIt.value()->type() == ChangeType::Move ||
+							changeIt.value()->type() == ChangeType::Deletion);
+
+				// Update the old Label according to labelNone(As per base)
+				changeIt.value()->oldLabel_ = labelNone;
+
+				// Update flags according to labels
+				if (changeIt.value()->newLabel() == changeIt.value()->oldLabel())
```

Comment:
It will be, but we want to keep it that way. Your code will potentially set it to Label.